### ISSUE TITLE
404 updating recurrence

### ISSUE THREAD

**Issue by jmfarina** [nutso/redmine-plugin-recurring-tasks#70]
While editing a recurrence, attempting to save the changes raises a 404 due to the following error:
ActionController::RoutingError (No route matches [PUT] "/projects/46/recurring_tasks/2/update")

The following patch fixes the issue by changing the update handler from POST to PUT:

```ruby
diff --git a/config/routes.rb b/config/routes.rb
index 926863e..23edba9 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -9,5 +9,5 @@
 match 'projects/:project_id/recurring_tasks/create', :to => 'recurring_tasks#create', :via => 'post'
 match 'projects/:project_id/recurring_tasks/:id', :to => 'recurring_tasks#show', :as => :recurring_task, :via => 'get'
 match 'projects/:project_id/recurring_tasks/:id/edit', :to => 'recurring_tasks#edit', :as => :edit_recurring_task, :via => 'get'
-match 'projects/:project_id/recurring_tasks/:id/update', :to => 'recurring_tasks#update', :via => [:get, :post, :patch]
+match 'projects/:project_id/recurring_tasks/:id/update', :to => 'recurring_tasks#update', :via => [:get, :put, :patch]
 match 'projects/:project_id/recurring_tasks/:id/destroy', :to => 'recurring_tasks#destroy', :as => :destroy_recurring_task, :via => [:post, :delete]
```

**Comment by marczona** [comment]
Works fine for me, thanks a lot!

**Comment by bcorepcs** [comment]
May I ask what your cron job looks like?

Brian Wajerski
Core Computer Solutions
brian@corepcs.com<mailto:brian@corepcs.com>
Help desk:414-727-1240
Direct: 414-727-3266
[cid:image001.png@01CFE9E3.73DB8CD0]
Technology Solutions For Your Business

**Comment by jmfarina** [comment]
Actually, I've used Rufus Scheduler instead of cron. Refer to https://github.com/nutso/redmine-plugin-recurring-tasks/issues/72 for further details.

**Comment by nutso** [comment]
Confirmed bug on Redmine 2.x; not an issue on Redmine 3.x. Scheduling method should have no impact on whether an update is successful or not.

**Comment by nutso** [comment]
FYI, the stable version 1.4 should not be impacted by this as it was prior to upgrading the plugin to support Redmine 3. (This impacts the current master and dev branches)

**Comment by nutso** [comment]
@jmfarina thanks for posting the patch; I fixed the update for v1.5 and destroy routes so works both Redmine 2.x and Redmine 3.x
